### ISSUE TITLE
SwitchLastCaseIsDefault rule regression for fallthrough before default case

### ISSUE THREAD

**Issue by patrickjmccarty** [SonarOpenCommunity/sonar-cxx#884]
After I upgraded the cxx plugin from 0.9.4 to 0.9.5, I am now getting incorrect issues reported from the `cxx:SwitchLastCaseIsDefault` rule. It is incorrectly reporting that I must "Add a default case to this switch", for switch statements that already have a default case as the last case, but are immediately preceded by a fallthrough case.

As a simple example:
```
switch (value)
{
  case 0:
    return 0;
  case 1: /* fallthrough */
  default:
    return -1;
}
```
I see no reason why code like this should be flagged as a defect, so I'm guessing it was simply an oversight.

The related issues I found when searching are: #320 and #381

**Comment by Bertk** [comment]
I did a short query and there is another commit involved #743. By the way, there is another issue with the C++ standard implementation which demands

>  Case labels and default labels shall occur only in switch statements. 

but the parser accept this 
```cpp
int main() {

//	switch (value)
	{
	  case 0:
		return 0;
	  case 1: /* fallthrough */
	  default:
		return -1;
	}
}
```

**Comment by guwirth** [comment]
Will have a look but have less time at the moment.

**Comment by guwirth** [comment]
@patrickjmccarty please try with latest snapshot and give us feedback.

**Comment by patrickjmccarty** [comment]
@guwirth I just verified with the latest sonar-cxx-plugin-0.9.7-SNAPSHOT.jar from AppVeyor build 421 and this rule seems to work properly now. Thanks!

My only other feedback would be that the advice says: "This clause should either take appropriate action or contain a suitable comment as to why no action is taken." But no compliant example is given for such a no-action comment, so it's not clear whether users can avoid writing default if they have a comment instead, or whether the default is always required, but should contain a comment if the only thing it does is break out. I assume you mean the latter, but you might clarify the advice or add an example like:

```
switch (state) {
  case 0:
  case 1:
    printf("0 or 1!");
    break;
  default:                             // Compliant
    // Don't print other values.       // Explanatory comment recommended for why no action is taken.
    break;
}
```
